float32 x_top_left
float32 y_top_left
float32 width
float32 height